### PR TITLE
Added check for backup driver in rke.go

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
Added backup driver validation in rke as other projects like pwx does not require rke client addition
**Which issue(s) this PR fixes** (optional)
Closes #https://portworx.atlassian.net/browse/PTX-20719
https://portworx.atlassian.net/browse/PA-1760

**Special notes for your reviewer**:
RKE with cluster provider : [https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-syste[…]/job/s3/job/px-backup-system-test-rke/333/consoleFull](https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/s3/job/px-backup-system-test-rke/333/consoleFull)
s3: [https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-syste[…]/s3/job/px-backup-system-test-vanilla/502/consoleFull](https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/s3/job/px-backup-system-test-vanilla/502/consoleFull)
RKE without cluster provider as rke : https://jenkins.pwx.dev.purestorage.com/job/Users/job/Snigdha/job/rke-s3-pipeline/9/consoleFull

### CODE REVIEW

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: drivers/scheduler/rke/rke.go [hunk 1]
```diff
@@ -44,20 +45,25 @@ func (r *Rancher) String() string {
...
-		URL:       rkeParametersValue.Endpoint,
-		TokenKey:  rkeParametersValue.Token,
-		AccessKey: rkeParametersValue.AccessKey,
-		SecretKey: rkeParametersValue.SecretKey,
-		Insecure:  true,
-	}
-	r.client, err = rancherClient.NewClient(&rancherClientOpts)
-	if err != nil {
-		return err
+	if flag.Lookup("backup-driver").Value.(flag.Getter).Get().(string) == "pxb" {
```
Comment (kshithijiyer-px) [inline drivers/scheduler/rke/rke.go:51]: Please add a comment here

Comment (snigdha-px) [reply drivers/scheduler/rke/rke.go:51]: done

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Review by ak-px — APPROVED** [review]
LGTM

File: drivers/scheduler/rke/rke.go [hunk 2]
```diff
@@ -44,20 +45,26 @@ func (r *Rancher) String() string {
...
-		TokenKey:  rkeParametersValue.Token,
-		AccessKey: rkeParametersValue.AccessKey,
-		SecretKey: rkeParametersValue.SecretKey,
-		Insecure:  true,
-	}
-	r.client, err = rancherClient.NewClient(&rancherClientOpts)
-	if err != nil {
-		return err
+	// This is needed only in case of px-backup as other platforms do not create new RKE client
+	if flag.Lookup("backup-driver").Value.(flag.Getter).Get().(string) == "pxb" {
```
Comment (mkoppal-px) [inline drivers/scheduler/rke/rke.go:52]: Minor: Use the const for `pxb` instead of string

**Review by mkoppal-px — APPROVED** [review]
LGTM except for one minor comment

File: drivers/scheduler/rke/rke.go (hunk 2 above)
Comment (snigdha-px) [reply drivers/scheduler/rke/rke.go]: done

**Pull request merged**